aspects/aspects.go

```diff
@@ -306,57 +328,142 @@ func (a *Aspect) Set(databag DataBag, request string, value interface{}) error {
 		return a.bundle.schema.Validate(data)
 	}
 
-	return &NotFoundError{
-		Account:    a.bundle.Account,
-		BundleName: a.bundle.Name,
-		Aspect:     a.Name,
-		Field:      request,
-		Cause:      "field not found",
+	return notFoundErrorFrom(a, request, "no matching write rule")
+}
+
+// Get returns the aspect value identified by the request. If either the named
+// aspect or the corresponding value can't be found, a NotFoundError is returned.
+func (a *Aspect) Get(databag DataBag, request string, value *interface{}) error {
+	matches, err := a.matchGetRequest(request)
+	if err != nil {
+		return err
+	}
+
+	var merged interface{}
+	for _, match := range matches {
+		var val interface{}
+		if err := databag.Get(match.storagePath, &val); err != nil {
+			if errors.Is(err, PathNotFoundError("")) {
+				continue
+			}
+			return err
+		}
+
+		// build namespace around result
+		namespace := match.namespace
+		for i := len(namespace) - 1; i >= 0; i-- {
+			val = map[string]interface{}{namespace[i]: val}
+		}
+
+		// merge result with results from other matching rules
+		merged, err = merge(merged, val)
+		if err != nil {
+			return err
+		}
+	}
+
+	if merged == nil {
+		return notFoundErrorFrom(a, request, "no value for matching rules")
```

Comment:
maybe "matching rules retrieve no value" would be even clearer?  that's probably a question for native speakers

Reply:
@andrewphelpsj when you review, could you let us know your preference here? Thank you

Reply:
I think @pedronis's suggestion better implies that there is a rule that matches the request. The usage of `retrieve` might be a little confusing, though.

Maybe `matching rules map to no value`? I'm not totally sold on that, though.

Also, since we're merging multiple values, should it be `values`, plural?

Reply:
Taking both comments into account, I went with `matching rules don't map to any values` which hopefully is clearer